src/gtc/gtir_to_oir.py

```diff
@@ -69,33 +69,9 @@ def add_horizontal_execution(
     def visit_ParAssignStmt(
         self, node: gtir.ParAssignStmt, *, mask: oir.Expr = None, ctx: Context, **kwargs: Any
     ) -> None:
-        tmp = oir.Temporary(name=f"tmp_{node.left.name}_{node.id_}", dtype=node.left.dtype)
-
-        ctx.add_decl(tmp)
-
-        ctx.add_horizontal_execution(
-            oir.HorizontalExecution(
-                body=[
-                    oir.AssignStmt(
-                        left=oir.FieldAccess(
-                            name=tmp.name, offset=CartesianOffset.zero(), dtype=tmp.dtype
-                        ),
-                        right=self.visit(node.right),
-                    )
-                ],
-                mask=mask,
-            )
-        )
         ctx.add_horizontal_execution(
             oir.HorizontalExecution(
-                body=[
-                    oir.AssignStmt(
-                        left=self.visit(node.left),
-                        right=oir.FieldAccess(
-                            name=tmp.name, offset=CartesianOffset.zero(), dtype=tmp.dtype
-                        ),
-                    )
-                ],
+                body=[oir.AssignStmt(left=self.visit(node.left), right=self.visit(node.right))],
```

Comment:
Isn’t this still needed for temporary fields? Or is something like this illegal anyway?
```
tmp = tmp[1, 0, 0]
```

Reply:
good catch. and good question...

Reply:
Maybe at this point we should be pragmatic and disallow the case, which adds yet another rule.